--- xhp-1.4/xhp/Makefile~	2013-10-05 21:12:10.000000000 +0300
+++ xhp-1.4/xhp/Makefile	2013-10-05 21:13:14.761834909 +0300
@@ -1,7 +1,10 @@
+CXX = g++
+OPTFLAGS = -Wall -O3 
+
 ifdef DEBUG
 CPPFLAGS = -fPIC -ggdb -Wall -DDEBUG
 else
-CPPFLAGS = -fPIC -g -Wall -O3 -minline-all-stringops
+CPPFLAGS = -fPIC -g $(OPTFLAGS) -minline-all-stringops
 endif
 
 ifdef PROFILE
@@ -28,7 +31,7 @@
 	re2c -c -o $@ $<
 
 %.o: %.cpp
-	g++ -c $(CPPFLAGS) -o $@ $<
+	$(CXX) -c $(CPPFLAGS) -o $@ $<
 
 xhp_preprocess.o: xhp_preprocess.cpp scanner.lex.hpp parser.yacc.hpp
 parser.yacc.o: scanner.lex.hpp
@@ -39,8 +42,8 @@
 
 libxhp.so: code_rope.o scanner.lex.o parser.yacc.o fastpath.o xhp_preprocess.o
-	g++ -shared -Wl,-soname,libxhp.so -o libxhp.so $^
+	$(CXX) -shared -Wl,-soname,libxhp.so -o libxhp.so $^
 
 xhpize: xhpize.cpp libxhp.a
-	g++ $(CPPFLAGS) -o $@ $^
+	$(CXX) $(CPPFLAGS) -o $@ $^
 
 .PHONY: all clean tags
